src/ProfilePage/Api/profileApi.ts

```diff
@@ -0,0 +1,260 @@
+import { SetStateAction } from "react";
+import { octokit } from "../../../environment/apiKey";
+import {
+  BranchInfo,
+  Commit,
+  CommitData,
+  CommitItem,
+  LanguageData,
+  Parent,
+  ProfileItem,
+  RepoItem,
+} from "../../Models/interfaces";
+
+export const getCommits = async (
+  repoInfo: RepoItem,
+  setCommits: {
+    (value: SetStateAction<CommitData[]>): void;
+  },
+  sendUpCommits: { (commits: CommitItem[]): void }
+) => {
+  if (repoInfo) {
+    try {
+      const res = await octokit.request(
+        `GET https://api.github.com/repos/${repoInfo.full_name}/commits`
+      );
+
+      if (res.status === 200) {
+        const data = res.data;
+        const newCommits = data.map(
+          (commit: { sha: string; commit: Commit; parents: Parent[] }) => ({
+            sha: commit.sha,
+            commit: commit.commit,
+            parents: commit.parents,
+          })
+        );
+        setCommits(newCommits);
+        sendUpCommits(newCommits);
```

Comment:
Please could you explain why you have 2 callback functions that do the same thing?

Reply:
Did you change this now to do both from the caller?